src/app/shipping/SingleShippingForm.tsx

```diff
@@ -119,6 +119,10 @@ class SingleShippingForm extends PureComponent<SingleShippingFormProps & WithLan
         const PAYMENT_METHOD_VALID = ['amazon', 'amazonpay'];
         const shouldShowBillingSameAsShipping = !PAYMENT_METHOD_VALID.some(method => method === methodId);
 
+        if (consignments.length > 1) {
+            this.handleAddressSelect(consignments[0].shippingAddress);
```

Comment:
I don't think it's a good idea to perform any updates inside a `render` function. `render` should just focus on returning the React elements that need rendering whenever there's a change in state. It shouldn't make any API calls or call `setState`, which is what `handleAddressSelect` is doing.

A separate question is, how come updating the address of the first consignment can cause all the other consignments collapse into one 🤔

Reply:
Thank you for the feedback. I have moved it to `componentDidMount()`.
All tests are passed now without modifications.

Reply:
What about the second question DC mentioned above?

Reply:
@animesh1987 The video is a recording of the current situation. The bug only appears when a user switches back to the single shipping-form and click `continue` without any interactions with address dropdown menu.

The bugfix acts the same as choosing the first option in the single shipping-form's dropdown menu. It does not change the current consignment update logic.

https://user-images.githubusercontent.com/88361607/133966139-06b5945e-3494-450f-9514-02dd75bb6508.mov